sql/12/cwvs-by-wordpress-version.sql

```diff
@@ -0,0 +1,92 @@
+# HTTP Archive query to get Core Web Vital "good" rates by WordPress version.
+#
+# WPP Research, Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See https://github.com/GoogleChromeLabs/wpp-research/pull/18
+
+CREATE TEMP FUNCTION IS_GOOD(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64)
+RETURNS BOOL
+AS (
+  good / (good + needs_improvement + poor) >= 0.75
+);
+
+CREATE TEMP FUNCTION IS_NON_ZERO(good FLOAT64, needs_improvement FLOAT64, poor FLOAT64)
+RETURNS BOOL
+AS (
+  good + needs_improvement + poor > 0
+);
+
+SELECT *
+FROM
+  (
+    SELECT
+      REGEXP_EXTRACT(info, '(\\d.\\d).*') AS major_version,
+      client,
+      COUNT(DISTINCT url) AS origins,
+      COUNT(DISTINCT IF(good_fid, url, NULL)) AS origins_with_good_fid,
+      COUNT(DISTINCT IF(good_cls, url, NULL)) AS origins_with_good_cls,
+      COUNT(DISTINCT IF(good_lcp, url, NULL)) AS origins_with_good_lcp,
+      COUNT(DISTINCT IF(any_fid, url, NULL)) AS origins_with_any_fid,
+      COUNT(DISTINCT IF(any_cls, url, NULL)) AS origins_with_any_cls,
+      COUNT(DISTINCT IF(any_lcp, url, NULL)) AS origins_with_any_lcp,
+      COUNT(DISTINCT IF(good_cwv, url, NULL)) AS origins_with_good_cwv,
+      COUNT(DISTINCT IF(any_lcp AND any_cls, url, NULL)) AS origins_eligible_for_cwv,
+      SAFE_DIVIDE(COUNTIF(good_cwv), COUNTIF(any_lcp AND any_cls))
```

Comment:
Why does this consider LCP and CLS, but not FID?

Reply:
I believe because a good portion of the data doesn't include FID measurement so it is excluded here. I'll track down  the conversation, I had the same question for Rick.

Reply:
Note that the `good_cwv` counted here **does** include a fid component (when available) - 
https://github.com/GoogleChromeLabs/wpp-research/pull/18/files#diff-a8114800a4698295ca813e56f73e31865bca68149f479c202971f52db959506dR60-R62

My memory of the explanation of the logic here is from an email thread with Rick:
"If you're trying to measure % of experiences with good LCP, I think it'd be ok to omit null LCP values. However there isn't really a corresponding "% of experiences with good LCP, FID, and CLS" because each metrics' distribution is independent. We don't necessarily know that an individual experience that had a good LCP also had a good CLS. So instead we evaluate the entire page/origin as having good CWV if 75%+ of each metrics' experiences are good. If there is no FID data (as is the case for ~30% of origins) then you can still evaluate the page/origin based on the other two CWV."

Also, see this HTTPArchive query where a similar calculation is used:

https://github.com/HTTPArchive/cwv-tech-report/blob/main/sql/monthly.sql#L209